lib/ddtrace.rb

```diff
@@ -17,6 +17,7 @@
 require 'ddtrace/metrics'
 require 'ddtrace/auto_instrument_base'
 require 'ddtrace/profiling'
+require 'datadog/security/autoload'
```

Comment:
Would it make sense to call this just `datadog/security`? If we always require this file from `ddtrace.rb`, then how is "autoload" different from most other files in the project?

(I'd call out that we do have `ddtrace/auto_instrument` and `datadog/profiling/preload` but those are actually NOT required automatically, and customers need to opt into them by actually adding those requires somewhere)

Reply:
I don't quite like the naming here, there's certainly some shuffling around to be done. Even `security/auto_instrument` doesn't quite behave like the tracer's one.

`autoload` is doing some magic so that `datadog/security` doesn't even appear unless an env var is set, maximising guarantees that appsec has no effect unless either the env var is set or `datadog/security` is manually required by the user with heightened intent. I did not want that to sit right into `datadog/security` so as to make this function isolated, and be able to easily prune it when it's not required anymore.

Reply:
Noted, and will attempt to address.

Reply:
Perhaps `datadog/security/try_load` or `datadog/security/load_if_enabled` or `datadog/security/maybe_load`, possibly with a note on the file itself explaining why the non-standard name?